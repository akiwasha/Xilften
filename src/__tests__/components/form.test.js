import React from 'react';
import { render } from '@testing-library/react';
import { Form } from '../../components';

jest.mock('react-router-dom');

describe('<Form />', () => {
  it('renders the <Form /> with the populated data', () => {
    const { container, getByText, queryByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>S'identifier</Form.Title>

        <Form.Base>
          <Form.Input placeholder="Adresse email" onChange={() => {}} />
          <Form.Input
            type="password"
            placeholder="Mot de passe"
            onChange={() => {}}
          />
          <Form.Submit disabled type="submit">
            S'identifier maintenant
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          Première visite sur xilften?{' '}
          <Form.Link to="signup">Inscrivez-vous.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          Cette page est protégée par Google reCAPTCHA pour nous assurer que
          vous n'êtes pas un robot.
        </Form.TextSmall>
      </Form>
    );

    expect(
      getByText(
        "Cette page est protégée par Google reCAPTCHA pour nous assurer que vous n'êtes pas un robot."
      )
    ).toBeTruthy();
    expect(getByText("S'identifier")).toBeTruthy();
    expect(getByText("S'identifier maintenant").disabled).toBeTruthy();
    expect(getByPlaceholderText('Adresse email')).toBeTruthy();
    expect(getByPlaceholderText('Mot de passe')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Form /> with an error', () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Votre adresse email est déjà utilisée</Form.Error>
        <Form.Submit type="submit">S'identifier</Form.Submit>
      </Form>
    );

    expect(getByText('Votre adresse email est déjà utilisée')).toBeTruthy();
    expect(queryByText("S'identifier").disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
