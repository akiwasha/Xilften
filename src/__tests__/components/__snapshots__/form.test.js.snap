// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with an error 1`] = `
<div
  class="form__Container-sc-17alsyo-0 fOtLPC"
>
  <div
    class="form__Error-sc-17alsyo-1 uYcEp"
  >
    Votre adresse email est déjà utilisée
  </div>
  <button
    class="form__Submit-sc-17alsyo-8 ddPkDN"
    type="submit"
  >
    S'identifier
  </button>
</div>
`;

exports[`<Form /> renders the <Form /> with the populated data 1`] = `
<div
  class="form__Container-sc-17alsyo-0 fOtLPC"
>
  <h1
    class="form__Title-sc-17alsyo-3 inyZHw"
  >
    S'identifier
  </h1>
  <form
    class="form__Base-sc-17alsyo-2 iKGYYP"
  >
    <input
      class="form__Input-sc-17alsyo-7 fuggHQ"
      placeholder="Adresse email"
    />
    <input
      class="form__Input-sc-17alsyo-7 fuggHQ"
      placeholder="Mot de passe"
      type="password"
    />
    <button
      class="form__Submit-sc-17alsyo-8 ddPkDN"
      disabled=""
      type="submit"
    >
      S'identifier maintenant
    </button>
  </form>
  <p
    class="form__Text-sc-17alsyo-4 jbCNfx"
  >
    Première visite sur xilften?
     
  </p>
  <p
    class="form__TextSmall-sc-17alsyo-5 vHjKB"
  >
    Cette page est protégée par Google reCAPTCHA pour nous assurer que vous n'êtes pas un robot.
  </p>
</div>
`;
